components/filecoin/rs/src/signature.rs

```diff
@@ -153,35 +109,40 @@ fn transaction_sign_bls_raw(
 /// * `unsigned_message_api` - an unsigned filecoin message
 /// * `private_key` - a `PrivateKey`
 ///
-/// https://github.com/Zondax/filecoin-signing-tools/blob/2e7b005b4733761de11d4a252cb481ca5aedb029/signer/src/lib.rs#L319
+/// https://github.com/Zondax/filecoin-signing-tools/blob/222362ae38a1ccd8523ea026d5298d4718b43e2b/signer-npm/src/lib.rs#L342
 pub fn transaction_sign_raw(
-    unsigned_message_api: &UnsignedMessage,
+    message: &UnsignedMessage,
     private_key: &PrivateKey,
 ) -> Result<Signature, SignerError> {
     // the `from` address protocol let us know which signing scheme to use
-    match unsigned_message_api.from.protocol() {
-        Protocol::Secp256k1 => {
-            let signed = transaction_sign_secp56k1_raw(unsigned_message_api, private_key)?;
-            Ok(Signature::SignatureSECP256K1(signed))
+    let signature = match message.from.protocol() {
+        fvm_shared::address::Protocol::Secp256k1 => {
+            transaction_sign_secp56k1_raw(message, private_key)?
         }
-        Protocol::BLS => {
-            let signed = transaction_sign_bls_raw(unsigned_message_api, private_key)?;
-            Ok(Signature::SignatureBLS(signed))
+        fvm_shared::address::Protocol::BLS => transaction_sign_bls_raw(message, private_key)?,
+        _ => {
+            return Err(SignerError::GenericString("Unknown signing protocol".to_string()));
         }
-        _ => Err(SignerError::GenericString("Unknown signing protocol".to_string())),
-    }
+    };
+
+    Ok(signature)
 }
 
-pub fn transaction_sign(transaction: &str, private_key: &[u8]) -> String {
-  let mut de = serde_json::Deserializer::from_str(transaction);
-  let mut sk = PrivateKey([0; SECRET_KEY_SIZE]);
-  sk.0.copy_from_slice(&private_key[..SECRET_KEY_SIZE]);
-  let message_user_api = MessageAPI::deserialize(&mut de);
-  if let Ok(message_user_api) = message_user_api {
-    let raw_signature = transaction_sign_raw(&message_user_api, &sk);
-    if let Ok(raw_signature) = raw_signature {
-        return base64::encode(raw_signature.as_bytes());
+pub fn transaction_sign(is_mainnet: bool, transaction: &str, private_key: &[u8]) -> String {
+    if (is_mainnet) {
+        set_current_network(Network::Mainnet);
+    } else {
+        set_current_network(Network::Testnet);
+    }
+    let mut de = serde_json::Deserializer::from_str(transaction);
+    let mut sk = PrivateKey([0; SECRET_KEY_SIZE]);
+    sk.0.copy_from_slice(&private_key[..SECRET_KEY_SIZE]);
+    let message_user_api = MessageAPI::deserialize(&mut de);
+    if let Ok(message_user_api) = message_user_api {
+        let raw_signature = transaction_sign_raw(&message_user_api, &sk);
+        if let Ok(raw_signature) = raw_signature {
+            return base64::encode(raw_signature.bytes());
+        }
     }
-  }
-  String::new()
+    String::new()
```

Comment:
While we're here, maybe comment that cxx doesn't support Result types, so return an empty string to signal failure.